aether-kernel/aether/kernel/api/views.py

```diff
@@ -391,6 +420,40 @@ def skeleton(self, request, pk=None, *args, **kwargs):
             'name': schema.schema_name,
         })
 
+    @action(detail=False, methods=['post'], url_path='unique-usage')
+    def unique_usage(self, request, *args, **kwargs):
+        '''
+        Given a list of mapping ids, this view checks if a schema used in
+        any of the supplied mappings is used outside the context of the mappings
+        supplied
+
+        Reached through POST '/schemas/unique-usage/'
+
+        expected payload: [
+            'mapping-1-uuid',
+            'mapping-2-uuid',
+            ...
+        ]
+
+        returned result: {
+            # object with properties keys as schemas used in the context of supplied mappings
+            # with values True|False
+            # True if the schema is used only the the context and nowhere else
+            # False if the schema is used by some other mapping
+            # e.g
+
+            Person : true
+
+            # indicating the schema 'Person' is used by one or more
+            # of the supplied mappings only
+        }
+        '''
+        mappings = request.data
```

Comment:
Being :smiling_imp: 

We need to check that the list of mapping ids belong to the current realm if multitenancy is enabled... :man_facepalming: 

```python
from django_eha_sdk.multitenancy.utils import filter_by_realm

mappings = filter_by_realm(self.request, 
                           model.Mapping.objects.filter(pk__in=request.data or []), 
                           'mappingset__project').values_list('id', flat=True)
```

Reply:
Why do we need to do that? 🤔

Reply:
Because if multi-tenancy is enabled you need to ensure that nobody is trying to get schemas form mappings that belong to different tenants.

Schemas are shared among all tenants but not the mappings/contracts.